crates/aptos/src/move_tool/lint.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::IncludedArtifactsArgs;
+use crate::common::types::{
+    CliCommand, CliError, CliTypedResult, MovePackageDir, OptimizationLevel,
+};
+use aptos_framework::{BuildOptions, BuiltPackage};
+use async_trait::async_trait;
+use clap::Parser;
+use move_compiler_v2::Experiment;
+use move_model::metadata::{CompilerVersion, LanguageVersion};
+
+/// Run a Lint tool to show additional warnings, in addition to ordinary warnings and/or
+/// errors generated by the Move 2 compiler.  Currently this is done by running the
+/// Move 2 compiler with a special pass, with optimizations disabled, but in the future
```

Comment:
If the all the optimizations are disabled, and compiler v2 is run, then code gen may fail because of too many variables in a function.

Reply:
Ok, I use the default options (or those specified on the command line), except for requiring compiler-v2.